artifactory.py

```diff
@@ -245,6 +245,12 @@ def sha256sum(filename):
     return sha256.hexdigest()
 
 
+def datetime_xheader(datetime):
```

Comment:
The function `datetime_xheader` doesn't work with headers, it's a serializer.
`if isinstance(datetime, datetime.datetime):` doesn't it lead to a bug? because `datetime` is not a module anymore, it's a function argument `datetime_xheader(datetime)`

I think if we return `int`, not `str` - it works in headers too.

Could you add a test on this?

Reply:
I must confess that I've not checked it, you're right the naming clash will likely mess this up, do you think it's necessary to allow the string version? I only used a datetime conversion at my code.

This is the code I've done as a hack, in which I've tried and the header works in setting up the created/last-modified date on the server:

```
def datetime_header(datetime):
    return str(int(round(1000*datetime.timestamp())))

# Custom version to allow change of Created and Last-Modified
def deploy_file(target, file_name, created, last_modified, parameters):
    md5 = md5sum(file_name)
    sha1 = sha1sum(file_name)
    sha256 = sha256sum(file_name)
    target = target / pathlib.Path(file_name).name

    with open(file_name, "rb") as fobj:
        url = str(target)

        matrix_parameters = (
            f";{encode_matrix_parameters(parameters)}" if parameters else None
        )
        headers = {}

        headers["X-Checksum-Md5"] = md5
        headers["X-Checksum-Sha1"] = sha1
        headers["X-Checksum-Sha256"] = sha256
        headers["X-Artifactory-Created"] = datetime_header(created)
        headers["X-Artifactory-Last-Modified"] = datetime_header(last_modified)

        target._accessor.rest_put_stream(
            url,
            fobj,
            headers=headers,
            session=target.session,
            verify=target.verify,
            cert=target.cert,
            timeout=target.timeout,
            matrix_parameters=matrix_parameters,
        )
```